### DISCUSSION TITLE
ARM (Apple Silicon) bottles

### DISCUSSION THREAD

**Discussion by jimtut** [Homebrew/discussions#246] [Everyday usage] [answered]
Sorry, pretty new to homebrew here, but I've been compiling open-source (the hard way!) for years, so not new to macOS and *nix systems.

I have a new M1 Mac mini, and most of my installs (from formulae?) are fine.  There's a known issue around openjdk, and that's preventing me installing my favorite tool, ffmpeg.

So, I've been reading about bottles, and how they are binary builds, and that they've started to build ARM-architecture bottles now.  OK, cool.  But, how can I tell if there's a bottle for ffmpeg (or any brew) for my architecture?

P.S.  It's not there yet, but hoping there's a way I can tell without just typing "brew install ffmpeg" and hoping!

**Comment by SMillerDev** [comment, marked as answer]
https://formulae.brew.sh/formula/ffmpeg lists the platforms that have bottles.

**Reply by jimtut** [reply]
According to another comment, I'm looking for a bottle with the label "arm64_big_sur" or something, in addition to the current "big_sur", etc, bottles.

**Comment by jimtut** [comment]
I see the list of OSes (including mine, Big Sur), but I don't see anything the differentiates between CPU architectures.  I don't think that was an issue until the M1, and I wouldn't have asked, but I know from an [open GitHub issue](https://github.com/Homebrew/brew/issues/7857#issuecomment-740132644) that ARM bottles are starting to come out, just trying to figure out how to tell.

Maybe the UI hasn't caught up with the labelling yet, or maybe only the bottles that support ARM are listing it (and the absence implies Intel?).

**Reply by carlocab** [reply]
If it says nothing about the architecture, then there is no ARM bottle. If it has an ARM bottle, it'll say something like arm64_big_sur, along with big_sur, catalina, etc.

There were a handful (~170) of formulae bottled in the past few days for ARM, but I think (all) the bottles have had to be withdrawn due to some unexpected issues with them.

**Reply by jimtut** [reply]
Thanks, that's what I was looking for!

BTW, I found a tap that provides a customizable ffmpeg, so I can build it without openjdk, so I'm back in business!  :-)

**Reply by lakemike** [reply]
Hi,
Please note there is a custom build for ffmpeg on Apple Silicon. It can be found at https://github.com/ssut/ffmpeg-on-apple-silicon
On my end and with one specific use case, this is 3x faster than ffmpeg provided by v3.0.
(Just saying and please not intended as critical comment, I understand the heroic efforts behind homebrew for M1...)
Michael

**Reply by carlocab** [reply]
Thanks, @lakemike. How does your build differ from the [Homebrew formula](https://github.com/Homebrew/homebrew-core/blob/98fd33463f970cb3a02978fe652d9bc2f5cf4ec0/Formula/ffmpeg.rb#L78)?

**Reply by lakemike** [reply]
Sorry, I should have been clearer that this is not my build and I cannot claim any credit for this. Please see above link and as well this guide: https://www.osxexperts.net/
